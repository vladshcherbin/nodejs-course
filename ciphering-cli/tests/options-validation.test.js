import validateOptions from '../src/options-validation'

test('Validate single option', async () => {
  await expect(validateOptions({ config: 'A' })).resolves.toStrictEqual({
    config: 'A'
  })
})

test('Validate multiple options', async () => {
  await expect(validateOptions({ config: 'A', input: 'input.txt' })).resolves.toStrictEqual({
    config: 'A',
    input: 'input.txt'
  })
})

test('Throw error when "config" option is missing', async () => {
  await expect(validateOptions({})).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "config" value is missing', async () => {
  await expect(validateOptions({ config: true })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "config" options has wrong value', async () => {
  await expect(validateOptions({ config: 'C1-B' })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "input" value is missing', async () => {
  await expect(validateOptions({ config: 'A', input: true })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "input" file is missing', async () => {
  await expect(validateOptions({ config: 'A', input: 'missing.txt' })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "input" file is a directory', async () => {
  await expect(validateOptions({ config: 'A', input: 'src' })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "output" value is missing', async () => {
  await expect(validateOptions({ config: 'A', output: true })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "output" file is missing', async () => {
  await expect(validateOptions({ config: 'A', output: 'missing.txt' })).rejects.toThrowErrorMatchingSnapshot()
})

test('Throw error when "output" file is a directory', async () => {
  await expect(validateOptions({ config: 'A', output: 'src' })).rejects.toThrowErrorMatchingSnapshot()
})
