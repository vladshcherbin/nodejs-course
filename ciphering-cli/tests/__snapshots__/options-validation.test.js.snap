// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throw error when "config" option is missing 1`] = `"\\"config\\" option is missing"`;

exports[`Throw error when "config" options has wrong value 1`] = `"\\"config\\" option contains wrong value"`;

exports[`Throw error when "config" value is missing 1`] = `"\\"config\\" option value is missing"`;

exports[`Throw error when "input" file is a directory 1`] = `"\\"input\\" file doesn't exist, no permission or is a directory"`;

exports[`Throw error when "input" file is missing 1`] = `"\\"input\\" file doesn't exist, no permission or is a directory"`;

exports[`Throw error when "input" value is missing 1`] = `"\\"input\\" option value is missing"`;

exports[`Throw error when "output" file is a directory 1`] = `"\\"output\\" file doesn't exist, no permission or is a directory"`;

exports[`Throw error when "output" file is missing 1`] = `"\\"output\\" file doesn't exist, no permission or is a directory"`;

exports[`Throw error when "output" value is missing 1`] = `"\\"output\\" option value is missing"`;
